File: app/src/main/java/com/arthur/gloria/logired/features/login/verifyemail/presentation/screen/VerifyEmailScreen.kt
package com.arthur.gloria.logired.features.login.verifyemail.presentation.screen

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Email
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.google.firebase.auth.FirebaseAuth
import kotlinx.coroutines.delay
import kotlinx.coroutines.tasks.await

@Composable
fun VerifyEmailScreen(
    email: String,
    onGoToLogin: () -> Unit
) {
    val green   = Color(0xFF1E7A5E)
    val bgColor = Color(0xFFF0F7F4)

    LaunchedEffect(Unit) {
        val auth = FirebaseAuth.getInstance()
        while (true) {
            delay(3000)
            try {
                auth.currentUser?.reload()?.await()
                if (auth.currentUser?.isEmailVerified == true) {
                    auth.signOut()
                    onGoToLogin()
                    break
                }
            } catch (e: Exception) {
            }
        }
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(bgColor)
            .padding(32.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Icon(
            Icons.Filled.Email,
            null,
            tint     = green,
            modifier = Modifier.size(80.dp)
        )

        Spacer(Modifier.height(24.dp))

        Text(
            "Verifica tu correo",
            fontSize   = 26.sp,
            fontWeight = FontWeight.Bold,
            color      = Color(0xFF1A1A1A)
        )

        Spacer(Modifier.height(12.dp))

        Text(
            "Enviamos un correo de verificación a:\n$email\n\nRevisa tu bandeja de entrada y haz clic en el enlace para activar tu cuenta.",
            fontSize   = 15.sp,
            color      = Color(0xFF7A9A8E),
            textAlign  = TextAlign.Center,
            lineHeight = 22.sp
        )

        Spacer(Modifier.height(32.dp))

        CircularProgressIndicator(color = green, modifier = Modifier.size(36.dp))

        Spacer(Modifier.height(16.dp))

        Text(
            "Esperando verificación...",
            fontSize = 13.sp,
            color    = Color(0xFF9A9A9A)
        )
    }
}

File: lib/features/login/verifyemail/presentation/verify_email_screen.dart
import 'package:flutter/material.dart';

class VerifyEmailScreen extends StatelessWidget {
  final String email;
  const VerifyEmailScreen({super.key, required this.email});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(24),
          child: Column(
            mainAxisSize: MainAxisSize.min,
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              const Icon(Icons.mark_email_unread_outlined,
                  size: 80, color: Colors.blue),
              const SizedBox(height: 24),
              const Text(
                'Verifica tu correo',
                style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
                textAlign: TextAlign.center,
              ),
              const SizedBox(height: 12),
              Text(
                'Te enviamos un enlace de verificación a $email.\nRevisa tu bandeja de entrada.',
                textAlign: TextAlign.center,
                style: const TextStyle(fontSize: 15, color: Colors.grey),
              ),
              const SizedBox(height: 32),
              ElevatedButton(
                onPressed: () => Navigator.pushNamedAndRemoveUntil(
                    context, '/login', (route) => false),
                child: const Text('Ir al login'),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
